src/prg_modelham_mod.F90

```diff
@@ -130,8 +130,10 @@ subroutine prg_twolevel_model(ea, eb, dab, daiaj, dbibj, dec, rcoeff, reshuffle,
     norbs = bml_get_N(h_bml)
     allocate(diagonal(norbs))
     allocate(row(norbs))
-    
-    call random_seed(seed)
+ 
+    seedin = seed    
+
+    call random_seed(PUT=seedin)
```

Comment:
I couldn't find an explanation what an array as see actually does. Could you explain?

Reply:
The put argument needs an array of integers to generate a good seed. But there seem to be a better way of doing this without having a fixed vector length.

Reply:
I am still confused :smile: Why not use something like [here](https://gcc.gnu.org/onlinedocs/gcc-5.2.0/gfortran/RANDOM_005fSEED.html)?

Reply:
@cnegre I also would like to get to the bottom of it. I see one issue too: what is the first call to random_seed() for? Otherwise it really looks like what @nicolasbock has posted ... (provided I understand Fortran well enough...)